### PR TITLE
MTDSA-17114 Changes to v2 Retrieve final declaration obligations

### PR DESCRIPTION
**Pull request by mattstephens-hmrc**

### CODE REVIEW

File: app/api/models/errors/CommonMtdErrors.scala [hunk 1]
```diff
@@ -26,6 +26,7 @@ object FromDateFormatError       extends MtdError("FORMAT_FROM_DATE", "The provi
 object ToDateFormatError         extends MtdError("FORMAT_TO_DATE", "The provided toDate is invalid", BAD_REQUEST)
 object StatusFormatError         extends MtdError("FORMAT_STATUS", "The provided status is invalid", BAD_REQUEST)
```
Comment (GazalNebHMRC) [inline app/api/models/errors/CommonMtdErrors.scala:27]: StatusFormatError and FormatStatusError are the same errors. One of them has to be removed, thanks.

File: app/v2/services/RetrieveCrystallisationObligationsService.scala [hunk 2]
```diff
@@ -36,15 +36,15 @@ class RetrieveCrystallisationObligationsService @Inject()(connector: RetrieveCry
...
+      downstreamResponseWrapper <- EitherT(connector.retrieveCrystallisationObligations(request)).leftMap(mapDownstreamErrors(errorMap))
       mtdResponseWrapper        <- EitherT.fromEither[Future](filterCrystallisationValues(downstreamResponseWrapper))
     } yield mtdResponseWrapper
 
     result.value
 
   }
 
-  private val downstreamErrorMap: Map[String, MtdError] =
+  private val errorMap: Map[String, MtdError] =
```
Comment (GazalNebHMRC) [inline app/v2/services/RetrieveCrystallisationObligationsService.scala:47]: Is there a specific reason the name was changed here? We usually call this val downstreamErrorMap in other API endpoints.

File: resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml [hunk 3]
```diff
@@ -63,11 +48,21 @@ get:
         </tr>
       </tbody>
     </table>
```
Comment (GazalNebHMRC) [inline resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml:50]: Should the description of Dynamic gov-test-scenario be added here or will the sandbox team add that in?

**Review by GazalNebHMRC — CHANGES_REQUESTED** [review]

File: resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml [hunk 4]
```diff
@@ -63,11 +48,21 @@ get:
...
+      description: |
+        The tax year the data applies to. The earliest permitted tax year is 2017-18.
+        
+        If a tax year is not specified then all obligation from CY-4 are returned. (CY = Current Tax Year)
+        eg. if current TY is 2023-24 then all obligations from 2019-20 till 2023-34 are returned.
+      schema:
+        type: string
+        example: 2022-23
+      required: false
+    - $ref: './common/queryParameters.yaml#/components/parameters/status'
```
Comment (GazalNebHMRC) [inline resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml:65]: The description for status query param should match the one finalised by Ben in the tech spec.

File: app/v2/services/RetrieveCrystallisationObligationsService.scala (hunk 2 above)
Comment (mattstephens-hmrc) [reply app/v2/services/RetrieveCrystallisationObligationsService.scala:47]: They're all meant to be called errorMap.

File: resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml (hunk 3 above)
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml:50]: yh, the sandbox team will be working on this next

File: resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml [hunk 5]
```diff
@@ -63,11 +48,21 @@ get:
...
 
   parameters:
     - $ref: './common/pathParameters.yaml#/components/parameters/nino'
-    - $ref: './common/queryParameters.yaml#/components/parameters/taxYear'
+    - name: taxYear
+      in: query
+      description: |
+        The tax year the data applies to. The earliest permitted tax year is 2017-18.
+        
+        If a tax year is not specified then all obligation from CY-4 are returned. (CY = Current Tax Year)
```
[the remark this answers is not in the document]
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml:59]: Yes, that's the idea

File: resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml (hunk 4 above)
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/retrieve_income_tax_crystallisation.yaml:65]: correct

**Review by GazalNebHMRC — APPROVED** [review]

**Pull request merged**